sys/auto_init/saul/auto_init_hts221.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2017 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/*
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization for HTS221 devices
+ *
+ * @author      Sebastian Meiling <s@mlng.net>
+ *
+ * @}
+ */
+
+#ifdef MODULE_HTS221
+
+#include "log.h"
+#include "saul_reg.h"
+#include "hts221.h"
+#include "hts221_params.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define HTS221_NUM    (sizeof(hts221_params)/sizeof(hts221_params[0]))
+
+/**
+ * @brief   Allocate memory for the device descriptors
+ */
+static hts221_t hts221_devs[HTS221_NUM];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[HTS221_NUM * 2];
+
+/**
+ * @brief   Reference the driver struct
+ * @{
+ */
+extern saul_driver_t hts221_saul_temp_driver;
+extern saul_driver_t hts221_saul_hum_driver;
+/** @} */
+
+
+void auto_init_hts221(void)
+{
+    for (unsigned i = 0; i < HTS221_NUM; i++) {
```

Comment:
Sorry, you also need to add an assert that checks the number of devices is the same as the number of saul info.